configure.py

```diff
@@ -192,6 +194,8 @@ def _run_command(self, cmdline):
 profilers = ['gmon', 'pprof']
 parser.add_option('--bootstrap', action='store_true',
                   help='bootstrap a ninja binary from nothing')
+parser.add_option('--bootstrap_from_amalgamation', action='store_true',
+                  help='bootstrap a ninja binary from the amalgamation')
```

Comment:
I think the option is required, unless you are ok with checking in `ninja-bootstrap.cc` in to this repo and updating it on every checkin.

The two cases are:

1. in the Ninja repo, there is no `ninja-bootstrap.cc` checked in, so `--bootstrap` will build `ninja-bootstrap.cc` from the other source files before building `ninja.bootstrap`. In this case, `ninja-bootstrap.cc` is considered to be an artifact.

2. in other repos, people will check in *only* `ninja-bootstrap.cc` and no other sources, so attempting to build `ninja-bootstrap.cc` will fail since the other sources aren't present. So users will pass `--bootstrap_from_amalgamation` because it's bootstrapping *directly* from the amalgamation, without trying to build the amalgamation first. In this case, `ninja-bootstrap.cc` is considered authoritative.

If you wanted to avoid the two cases (and the option), you would need to check `ninja-bootstrap.cc` into this repo and make sure it is updated every time someone makes a change to one of the other source files.